src/interface/common/middlewares/AuthorizationMiddleware.ts

```diff
@@ -99,11 +99,16 @@ export default class AuthorizationMiddleware {
 
         if (id)
             userDetailsInDb = await this._usersDetailsRepository.findOne({ userId: id });
+        const userInDb = await this._usersRepository.findOne({ userId: id });
+
+        userInDb.inProgress.status = 'waiting_question';
         if (email && !id) {
             const userInDb = await this._usersRepository.findOne({ email });
             userDetailsInDb = await this._usersDetailsRepository.findOne({
                 userId: userInDb.userId,
             });
+
+            userInDb.inProgress.status = 'waiting_question';
```

Comment:
_"Toda vez que a secret question for usada, o status do usuário deve mudar para o valor: waiting-question ( AuthorizationMiddleware )"_

Neste momento do código a secret question ainda não foi utilizada ela só vai ser utilizada na linha 122, e apenas após a linha 125 o status deve ser alterado, garantindo que uma vez que o usuário utilize sua secret question corretamente ele precise escolher uma nova secret question.

---

Outro ponto é que o novo status `waiting-question` está sendo definido mas não está sendo salvo no banco,  sem salvar não conseguimos mapear no sistema para pedir ao usuário que redefina sua secret question.